pkg/mesos/ip.go

```diff
@@ -0,0 +1,74 @@
+package mesos
+
+import (
+	"fmt"
+	"net"
+	"os"
+)
+
+// This helper function gives a Mesos container's IP address, when used from
+// the container's network namespace.
+//
+// The function first checks the presence of
+// `MESOS_CONTAINER_IP`  and than `LIBPROCESS_IP` (in that order).  The
+// expectation is that only one of these variables would be set.  Starting
+// Mesos 1.4 the `default-executor` setsup the container's IP address in
+// `MESOS_CONTAINER_IP`. However, prior to Mesos 1.4 if the container was
+// running on the host network the IP address would be set in `LIBPROCESS_IP
+// and if the container was running on a CNI network the `LIBPROCESS_IP` would
+// be set to 0.0.0.0 forcing a hostname resolution to resolv the container's
+// CNI IP address.
+func ContainerIP() (containerIP net.IP, err error) {
+	// NOTE: Make sure to maintain the order of the environment variables.
+	envs := [2]string{"MESOS_CONTAINER_IP", "LIBPROCESS_IP"}
+	for _, env := range envs {
+		ip := os.Getenv(env)
+		if ip != "" {
+			containerIP = net.ParseIP(ip)
+			if containerIP == nil {
+				err = fmt.Errorf("Invalid %s found: %s", env, ip)
+				return
```

Comment:
should this try to proceed to LIBPROCESS_IP if MESOS_CONTAINER_IP is invalid?

Reply:
No if `MESOS_CONTAINER_IP` is set we expect it to be valid else this a bug in Mesos.